lib/parser.js

```diff
@@ -91,32 +71,68 @@ class Parser extends EventEmitter {
         }
 
         this.includeSourceLocations = options.includeSourceLocations;
-        this.componentName = null;
+        this.componentName = options.componentName;
         this.template = options.source.template;
         this.filename = options.filename;
-        this.eventsEmmited = {};
+        this.eventsEmitted = options.eventsEmitted;

```

Comment:
Can you please describe changes of this kind (L74, L80, L81), I can't understand this

Reply:
I wanted to centralize the source of default values, so I put all of them in default options and then assign everything from the normalized options object (which contain all those keys). But now I realize it might not be related to external options at all. Should I separate the internal properties (`componentName`, `eventsEmitted`, `identifiers` and `imports`) from the user options? I also [asked in issue #29](https://github.com/alexprey/sveltedoc-parser/issues/29#issuecomment-738277884) if I should change the default of `options.defaultMethodVisibility` and `defaultActionVisibility` because they don't exist in options (I used `utils.DEFAULT_VISIBILITY` as default value).

I think I'll revert the changes.

Reply:
Ok, I understand. This default values should not be passed from user options, you can use additional object for that purpose.
Sorry, about default visibility states for methods and actions, I'm miss that. Looks like that feature was not documented, but I think that it's OK to use logically correct value for that. If user want to redefine default visibility for methods, we should think why this should be done by him? I think it can be usufully only when you wants to extract all methods or actions from component, but actually we have ignoreVisilibty option. So, yes you can change this behaviour and use default const value.

Reply:
I reverted most of the changes for the internal properties and separated the constructor in different sections to make things clearer. I also removed the added validations for those properties which where useless in the end.

Reply:
Thanks!